erdos/ros/ros_input_data_stream.py

```diff
@@ -61,13 +64,20 @@ def _on_msg(self, msg):
             # Also, maintain the lowest watermark observed.
             low_watermark = msg.timestamp
             for stream, watermark in self.op._stream_to_high_watermark.items():
-                if stream != msg.stream_name:
-                    if not watermark or watermark < msg.timestamp:
-                        return
-                    if low_watermark > watermark:
-                        low_watermark = watermark
+                # TODO(yika): big HACK on ignoring watermark sent on stream with label 'no_watermark' = true
+                if stream not in self.op._stream_ignore_watermarks:
+                    if stream != msg.stream_name:
+                        if not watermark or watermark < msg.timestamp:
+                            return
+                        if low_watermark > watermark:
+                            low_watermark = watermark
             msg = WatermarkMessage(low_watermark)
-            
+
+            # Checkpoint
+            if self.op._checkpoint_enable:
+                if msg.timestamp.coordinates[0] % self.op._checkpoint_freq == 0:
```

Comment:
Can we please call here a user provided lambda instead of hardcoding logic that check on the first timestamp coordinate? The lambda receives msg.timestamp, and thus the user write custom checkpointing policies. For example, in Pylot the timestamps are (simulation_time, sequence_number), and it could be beneficial to be able to use the sequence number to decide when to checkpoint.